### PR TITLE
docs: improve ignorePaths config option documentation

### PR DESCRIPTION
**Pull request by HonkingGoose**
## Changes:

- Make text more descriptive
- Add example of how to use `ignorePaths`
- Explain what `config:base` ignores for you when you extend from it

## Context:

I noticed somebody got very confused with the behavior of the `:ignoreModulesAndTests` preset:

- https://github.com/renovatebot/renovate/issues/11823

I checked our current docs for the `ignorePaths` config option, and it can use a bit of work to make things clearer.

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: docs/usage/configuration-options.md [hunk 1]
```diff
@@ -1078,8 +1078,28 @@ The above is the same as if you wrote this package rule:
 
 ## ignorePaths
 
-Using this setting, you can selectively ignore package files that you don't want Renovate autodiscovering.
-For instance if your repository has an "examples" directory of many package.json files that you don't want to be kept up to date.
+With this setting you can selectively ignore package files that would normally be "autodiscovered" and updated by Renovate.
```
Comment (rarkins) [inline docs/usage/configuration-options.md:1081]:
```suggestion
Renovate will extract dependencies from every file it finds in a repository, unless that file is explicitly ignored.
With this setting you can selectively ignore package files that would normally be "autodiscovered" and updated by Renovate.
```

File: docs/usage/configuration-options.md [hunk 2]
```diff
@@ -1078,8 +1078,28 @@ The above is the same as if you wrote this package rule:
...
+
+For instance if your repository has an `"**/examples/**"` directory with many `package.json` files that you want to ignore.
```
Comment (rarkins) [inline docs/usage/configuration-options.md:1083]:
```suggestion
For instance if you have a project with an `"**/examples/**"` directory you wish to ignore:
```

File: docs/usage/configuration-options.md [hunk 3]
```diff
@@ -1078,8 +1078,28 @@ The above is the same as if you wrote this package rule:
...
+
+<!-- TODO: Grabbed the ignorePaths array from our internal preset ignoreModulesAndTests located at: /lib/config/presets/internal/defaults.ts -->
+<!-- TODO: This example should probably be wrapped in a `packageRule` array? -->
+<!-- TODO: Not sure what the correct example should be. -->
+
```
Comment (rarkins) [inline docs/usage/configuration-options.md:1088]:
```suggestion
```

File: docs/usage/configuration-options.md [hunk 4]
```diff
@@ -1078,8 +1078,28 @@ The above is the same as if you wrote this package rule:
...
+  "ignorePaths": [
+    "**/node_modules/**",
+    "**/bower_components/**",
+    "**/vendor/**",
+    "**/examples/**",
+    "**/__tests__/**",
+    "**/test/**",
+    "**/tests/**",
+    "**/__fixtures__/**"
+  ]
```
Comment (rarkins) [inline docs/usage/configuration-options.md:1100]:
```suggestion
  "ignorePaths": ["**/examples/**"]
```

File: docs/usage/configuration-options.md [hunk 5]
```diff
@@ -1078,8 +1078,28 @@ The above is the same as if you wrote this package rule:
...
+}
+```
```
Comment (rarkins) [inline docs/usage/configuration-options.md:1103]:
```suggestion

Useful to know: Renovate's default ignore is `node_modules` and `bower_components` only, however if you are extending the popular `config:base` preset then it adds ignore patterns for `vendor`, `examples`, `test(s)` and `fixtures` directories too.

```

**Review by rarkins — CHANGES_REQUESTED** [review]

**Comment by HonkingGoose** [comment]
Thank you very much @rarkins! That is way better than my text! I've applied all your suggestions.

I wanted to add the explanation of what `config:base` ignores for you, but couldn't find the right words to say so. So thanks again for adding that explanation!

I think this is ready for a new round of review. 😉

**Review by rarkins — APPROVED** [review]

File: docs/usage/configuration-options.md [hunk 6]
```diff
@@ -1078,8 +1078,18 @@ The above is the same as if you wrote this package rule:
 
 ## ignorePaths
 
-Using this setting, you can selectively ignore package files that you don't want Renovate autodiscovering.
-For instance if your repository has an "examples" directory of many package.json files that you don't want to be kept up to date.
+Renovate will extract dependencies from every file it finds in a repository, unless that file is explicitly ignored.
+With this setting you can selectively ignore package files that would normally be "autodiscovered" and updated by Renovate.
+
+For instance if you have a project with an `"**/examples/**"` directory you wish to ignore:
```
Comment (rarkins) [inline docs/usage/configuration-options.md:1084]:
```suggestion
For instance if you have a project with an `"examples/"` directory you wish to ignore:
```

**Review by rarkins — CHANGES_REQUESTED** [review]

**Review by rarkins — APPROVED** [review]

**Pull request merged**